src/runtime_src/core/pcie/emulation/common_em/unix_socket.cxx

```diff
@@ -95,34 +105,37 @@ void unix_socket::start_server(double timeout_insec, bool fatal_error)
   }
 
   fd = accept(sock, 0, 0);
+  mpoll_on_filedescriptor = {fd, POLLERR, 0}; 
   close(sock);
   if (fd == -1){
     perror("socket acceptance failed");
     exit(1);
   } else {
     server_started.store(true);
+    simprocess_socket_live.store(true);
   }
   return;
 }
 
 ssize_t unix_socket::sk_write(const void *wbuf, size_t count)
 {
-  if (not server_started) {
-    std::cout<< "\n unix_socket::sk_write failed, no socket connection established.\n";
+  if (not server_started || (false == simprocess_socket_live) ) {
```

Comment:
A mix of everything, why not just `!`.  

Inconsistent use of store() and load().  I thought you mentioned (and I agreed) that maybe store and load make it explicit that we are dealing with atomicity, but the use is inconsistent.  Personally, I can live without store() and load().